dirty_cat/datasets/_fetching.py

```diff
@@ -162,6 +164,76 @@ def fetch_openml_dataset(
     }
 
 
+def _fetch_world_bank_data(
+    indicator_id: str,
+    indicator_name: str,
+    data_directory: Path = get_data_dir(),
+) -> Dict[str, Any]:
+    """
+    Gets a dataset from World Bank open data platform
+    (https://data.worldbank.org/).
+
+    Parameters
+    ----------
+    indicator_id: str
+        The ID of the indicator's dataset to fetch.
+    indicator_name: str
+        Name of the indicator to extract.
+    data_directory: Path
+        Optional. A directory to save the data to.
+        By default, the dirty_cat data directory.
+
+    Returns
+    -------
+    Dict[str, Any]
+        A dictionary containing:
+          - ``description``: str
+              The description of the dataset,
+              as gathered from World Bank data.
+          - ``source``: str
+              The dataset's URL from the World Bank data platform.
+          - ``path``: pathlib.Path
+              The local path leading to the dataset,
+              saved as a CSV file.
+
+    """
+    data_directory.mkdir(exist_ok=True, parents=True)
+    unzip_path = data_directory / "world_bank_unzipped_files"
+    data_directory = data_directory.resolve() / "world_bank.zip"
+    # Download the file :
+    url = f"https://api.worldbank.org/v2/en/indicator/{indicator_id}?downloadformat=csv"
+    urllib.request.urlretrieve(url, data_directory)
+    try:
+        # Extract csv from zip file :
+        with ZipFile(data_directory, "r") as file:
+            names = file.namelist()
+            for name in names:
+                if "Metadata" not in name:
+                    file.extract(name, path=unzip_path)
+                    true_file = name
+        file.close()
+    except BadZipFile:
+        raise FileNotFoundError(
+            f"Couldn't find csv file {data_directory!s}, {indicator_id} seems invalid."
+        )
+    # Read csv file
+    csv_path = Path(unzip_path) / true_file
+    df = pd.read_csv(csv_path, skiprows=3)
```

Comment:
As discussed IRL: Using a [`tempfile.TemporaryDirectory`](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory) context manager interacting with temporary directories directly is appropriate. See [this example](https://docs.python.org/3/library/tempfile.html#examples).